File: backend/schemas/billing.py
from datetime import datetime

from pydantic import BaseModel


class PlanDetail(BaseModel):
    name: str  # starter, pro, premium
    display_name: str
    price_cents: int  # 1900, 4900, 9900
    quota: int  # 25, 100, 500
    features: list[str]


class PlansResponse(BaseModel):
    plans: list[PlanDetail]


class SubscriptionStatus(BaseModel):
    plan: str | None = None  # None = no subscription
    status: str | None = None
    applications_used: int = 0
    quota: int = 0
    remaining: int = 0
    current_period_end: datetime | None = None
    is_whitelisted: bool = False


class CheckoutRequest(BaseModel):
    plan: str  # starter, pro, premium


class CheckoutSession(BaseModel):
    checkout_url: str
    session_id: str


class PortalSession(BaseModel):
    portal_url: str

File: frontend/src-v2/types/billing.ts
export interface PlanDetail {
  name: string;
  display_name: string;
  price_cents: number;
  quota: number;
  features: string[];
}

export interface PlansResponse {
  plans: PlanDetail[];
}

export interface SubscriptionStatus {
  plan: string | null;
  status: string | null;
  applications_used: number;
  quota: number;
  remaining: number;
  current_period_end: string | null;
  is_whitelisted: boolean;
}

export interface CheckoutSession {
  checkout_url: string;
  session_id: string;
}

export interface PortalSession {
  portal_url: string;
}

export interface BillingError {
  code: "no_subscription" | "quota_exceeded";
  message: string;
  plan?: string;
  quota?: number;
  applications_used?: number;
  next_plan?: string;
}
